### PR TITLE
[DO-1689][Contains migration] fixed sorting of last_employer and last_job_title fields

### PR DESCRIPTION
**Pull request by EmmaSamoilova**
improve Pep model: changed null=True to default=''
changed ordering: symbols should be first, empty string or '-' last

### CODE REVIEW

**Comment by criticus** [comment]
is this migration affecting existing records? Would their `updated_at` field be updated?

**Review by Andrii17s — APPROVED** [review]

**Comment by EmmaSamoilova** [comment]
> is this migration affecting existing records? Would their `updated_at` field be updated?
I applied this migration locally today. updated_at field was not changed and my records were not lost

**Review by roman-tiukh — CHANGES_REQUESTED** [review]
Look at this PR #1155

File: business_register/converter/pep.py [hunk 1]
```diff
@@ -97,10 +97,10 @@ def save_to_db(self, json_file):
             last_name = pep_dict['last_name'].lower()
             fullname = pep_dict['full_name'].lower()
             fullname_transcriptions_eng = pep_dict['names'].lower()
-            last_job_title = pep_dict.get('last_job_title')
+            last_job_title = pep_dict.get('last_job_title', '')
             if last_job_title:
                 last_job_title = last_job_title.lower()
-            last_employer = pep_dict.get('last_workplace')
+            last_employer = pep_dict.get('last_workplace', '')
             if last_employer:
```
Comment (roman-tiukh) [inline business_register/converter/pep.py:104]: PepJsonConverter not used now

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: data_ocean/filters.py [hunk 2]
```diff
@@ -50,3 +53,13 @@ def get_search_terms(self, request):
         params = request.query_params.get(self.search_param, '')
         params = params.replace('\x00', '')
         return [params]
+
+
+class NullLastOrderingFilter(filters.OrderingFilter):
+
+    def filter(self, qs, value):
+        NULL_LAST_PARAM = ('last_employer', 'last_job_title', 'last_employer_en', 'last_job_title_en')
```
Comment (roman-tiukh) [inline data_ocean/filters.py:61]: This configuration must be passed into `__init__` methods

File: data_ocean/filters.py [hunk 3]
```diff
@@ -50,3 +53,13 @@ def get_search_terms(self, request):
...
+        if value and len(value) == 1 and value[0] in NULL_LAST_PARAM:
+            qs = qs.annotate(param=NullIf(f'{value[0]}', Value(''))).order_by('param')
```
Comment (roman-tiukh) [inline data_ocean/filters.py:63]: param... param must be generated from value

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: data_ocean/filters.py (hunk 3 above)
Comment (EmmaSamoilova) [reply data_ocean/filters.py]: done

File: data_ocean/filters.py (hunk 2 above)
Comment (EmmaSamoilova) [reply data_ocean/filters.py]: done

File: data_ocean/filters.py [hunk 4]
```diff
@@ -50,3 +53,16 @@ def get_search_terms(self, request):
         params = request.query_params.get(self.search_param, '')
         params = params.replace('\x00', '')
         return [params]
+
+
+class NullLastOrderingFilter(filters.OrderingFilter):
+
+    def __init__(self, *args, **kwargs):
+        self.NULL_LAST_PARAM = kwargs.pop('null_last_param', ())
```
Comment (roman-tiukh) [inline data_ocean/filters.py:61]:
```suggestion
        self.null_last_fields = kwargs.pop('null_last_fields', ())
```

**Review by roman-tiukh — APPROVED** [review]

File: data_ocean/filters.py (hunk 4 above)
Comment (EmmaSamoilova) [reply data_ocean/filters.py]: done

**Review by Andrii17s — APPROVED** [review]

**Pull request merged**